pkg/virt-launcher/virtwrap/agent-poller/agent_poller_test.go

```diff
@@ -82,4 +84,38 @@ var _ = Describe("Qemu agent poller", func() {
 			Expect(agentStore.AgentUpdated).ToNot(Receive())
 		})
 	})
+
+	Context("PollerWorker", func() {
+		const fakeAgentCommandName = "foo"
+
+		It("executes the agent commands at least once", func() {
+			w := PollerWorker{
+				CallTick:      1,
+				AgentCommands: []AgentCommand{fakeAgentCommandName},
+			}
+			c := make(chan struct{})
+
+			var execCounter int
+			go w.Poll(func(commands []AgentCommand) { execCounter++ }, c)
```

Comment:
maybe worth to add mechanism that will ensure this go routine is finished ?
up to you

Reply:
The mechanism is in place, see the inputted channel.

Reply:
I see the channel, what tells you the function didnt hang?
Should the unit test make sure it gathers its go routines or you trust it ?

Reply:
The test closes the channel, and that assures the goroutine is terminated.
If you see a problem with it, present a specific/real scenario, general comments are not helpful.

Reply:
You trust the core code that implement w.Poll
if there will be a bug that the routine enter a deadlock  / endless loop you will not know that.
Its up to you if you want to trust it or test it.

Reply:
Raising such claims without some kind of proof/example/alternative is not helpful.
I am testing the code this goroutine is calling, so if it blocks it is a bug in production and it should show in the tests as we are suppose to check them. Expecting the correct value seems enough to me, but if you see a specific problem please raise an example and do not use generic comments.

Reply:
IMO the discussion here can can be generic, because its either the unit test should test its fine or assume its fine.
Answering this will suggest the AI.
But i have nothing more to add atm beside what was raised already, thanks.